Lib/test/test_format.py

```diff
@@ -272,45 +271,149 @@ def test_common_format(self):
-        test_exc_common('% %s', 1, ValueError,
-                        "unsupported format character '%' (0x25) at index 2")
+        test_exc_common('abc %', (), ValueError, "stray % at position 4")
+        test_exc_common('abc % %s', 1, ValueError,
+                        "stray % at position 4 or unexpected format character '%' at position 6")
+        test_exc_common('abc %z', 1, ValueError,
+                        "unsupported format %z at position 4")
+        test_exc_common("abc %Id", 1, ValueError,
+                        "unsupported format %I at position 4")
+        test_exc_common("abc %'d", 1, ValueError,
+                        "stray % at position 4 or unexpected format character ''' at position 5")
+        test_exc_common("abc %1 d", 1, ValueError,
+                        "stray % at position 4 or unexpected format character ' ' at position 6")
+        test_exc_common('abc % (x)r', {}, ValueError,
+                        "stray % at position 4 or unexpected format character '(' at position 6")
+        test_exc_common('abc %((x)r', {}, ValueError,
+                        "stray % or incomplete format key at position 4")
+        test_exc_common('%r %r', 1, TypeError,
+                        "not enough arguments for format string (got 1)")
+        test_exc_common('%r %r', (1,), TypeError,
+                        "not enough arguments for format string (got 1)")
+        test_exc_common('%r', (), TypeError,
+                        "not enough arguments for format string (got 0)")
+        test_exc_common('abc %' + '9'*50 + 'r', 1, ValueError,
+                        "width too big at position 4")
+        test_exc_common('abc %.' + '9'*50 + 'r', 1, ValueError,
+                        "precision too big at position 4")
+        test_exc_common('%r %*r', 1, TypeError,
+                        "not enough arguments for format string (got 1)")
+        test_exc_common('%r %*r', (1,), TypeError,
+                        "not enough arguments for format string (got 1)")
+        test_exc_common('%*r', (1,), TypeError,
+                        "not enough arguments for format string (got 1)")
+        test_exc_common('%*r', (), TypeError,
+                        "not enough arguments for format string (got 0)")
+        test_exc_common('%r %.*r', 1, TypeError,
+                        "not enough arguments for format string (got 1)")
+        test_exc_common('%r %.*r', (1,), TypeError,
+                        "not enough arguments for format string (got 1)")
+        test_exc_common('%.*r', (1,), TypeError,
+                        "not enough arguments for format string (got 1)")
+        test_exc_common('%.*r', (), TypeError,
+                        "not enough arguments for format string (got 0)")
+        test_exc_common('%(x)r', 1, TypeError,
+                        "format requires a mapping, not int")
+        test_exc_common('%*r', 3.14, TypeError,
+                        "* requires int, not float")
```

Comment:
Without context, it's uneasy to understand that the error comes from a format string. Maybe write `%*` instead?

```suggestion
                        "%* requires int, not float")
```

Reply:
It is perhaps better to remove this test, because the example is incorrect in more than one way. If `*` is used, then we need more than one argument, and in that case we will get "format argument N" in the error message, like in the following line..